src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -50,7 +54,16 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                try
+                {
+                    string envType = Environment.GetEnvironmentVariable(RequestTracingConstants.AspNetCoreEnvironmentVariable) ??
+                                        Environment.GetEnvironmentVariable(RequestTracingConstants.DotNetCoreEnvironmentVariable);
+                    if (envType.Equals(_environmentName, StringComparison.OrdinalIgnoreCase))
```

Comment:
.NET assumes that the environment is Production by default if these env var are not set. If we want to follow that pattern, we should still check if user specified environment name is Production and return a valid instance of AzureAppConfigurationProvider.